### ISSUE TITLE
Any way of using this library with a custom newQuery?

### ISSUE THREAD

**Issue by andrechalom** [lazychaser/laravel-nestedset#233]
This library works great for me with some simple models, but I need to reimplement `newQuery` on some models to add the result of some MySQL functions as columns, such as:

```php
public function newQuery($excludeDeleted = true)
{
return parent::newQuery($excludeDeleted)->addSelect('*', DB::raw('AsText(geom) geom, Area(geom) area'));
}
```

However, when I append these columns in the `newQuery`, trying to move nodes from one point in the tree to another results in this weird error, where the "rgt" column apparently gets mixed with the "name" column:

>  (1/1) ErrorException
A non-numeric value encountered
in QueryBuilder.php (line 555)
at HandleExceptions->handleError(2, 'A non-numeric value encountered', '/home/chalom/git/opendatabio/vendor/kalnoy/nestedset/src/QueryBuilder.php', 555, array('key' => 7, 'position' => 5, 'lft' => 7, 'rgt' => 'Amazonas', 'from' => 5, 'to' => 4))in QueryBuilder.php (line 555)
at QueryBuilder->moveNode(7, 5)in NodeTrait.php (line 590)
at Location->moveNode(5)in NodeTrait.php (line 572)

Is there anything I can do to fix this in `newQuery`? Or is this something that should be fixed on NestedSet? If it's the latter, I can try to write a PR, but I would need some pointers on where this needs fixing.

**Versions**: laravel-nestedset 4.2.7, Laravel 5.4.27

Best regards!

**Comment by andrechalom** [comment]
Digging into this issue, I noticed this error happens because the MoveNode method is using the first two columns as `($rgt, $lft)`. From what I understood from the code, these lines should match the actual left and right columns:
https://github.com/lazychaser/laravel-nestedset/blob/v4/src/QueryBuilder.php#L38-L39

However, this is not happening when I use a custom `newQuery`. I do not know why is this happening, but I have fixed this behavior in my code by explicitly including _lft and _rgt at the start of the query:

```php
public function newQuery($excludeDeleted = true)
{
    return parent::newQuery($excludeDeleted)->addSelect(
        '_lft',
        '_rgt',
        // other stuff
        DB::raw('AsText(geom) geom, Area(geom) area'));
}
```

I hope this can be of any use to you!
